### ISSUE TITLE
Add new place (not just Leeds Climate Innovation District) as a case study

### ISSUE THREAD

**Issue by Robinlovelace** [a-b-street/abstr#10]
Discussed today, it would be cool to work through the process of getting this in A/B Street: https://www.pct.bike/m/?r=isle-of-wight

Up for giving it a go, but probably not until late March or early April.

**Comment by natesheehan** [comment]
Yeah this would be cool addition, and would be great for linking between the pct / ab Street ecosystem. I can have a look at this for the meantime🤘

**Comment by Robinlovelace** [comment]
Fantastic you're up for taking a look @natesheehan. I'll change the title to broaden the scope.

**Comment by Robinlovelace** [comment]
A small town may be better than IoW, which has a few issues that I discovered in conversation with Dustin. Which town though? You choose! Key thing is minimal OSM data sizes and clear educational value - e.g. ~10 MSOAs and ~60k people should be fine.

**Comment by dabreegster** [comment]
IoW should be fine; the only unusual problem was that the water polygon didn't show up. I might have to do something special to handle coastlines for islands, but I'd be happy to look at it. Feel free to try the import instructions (https://a-b-street.github.io/docs/howto/new_city.html) and let me know what issues you hit.

**Comment by Robinlovelace** [comment]
Isn't it going to be a bit bit though? Advantage of IoW is the OSM data is small and self-contained so happy to go with that if it has upstream benefits for A/B Street - interesting to see the smallest zones in Geofabrik:




```
osmextract::geofabrik_zones[order(osmextract::geofabrik_zones$pbf_file_size), c("name", "pbf_file_size")][21:30, ]
Simple feature collection with 10 features and 2 fields
geometry type:  MULTIPOLYGON
dimension:      XY
bbox:           xmin: -8 ymin: -22.2411 xmax: 173.6089 ymax: 63
geographic CRS: WGS 84
               name pbf_file_size                       geometry
193        Maldives       3270259 MULTIPOLYGON (((71.7 7.69, ...
70          Comores       3367083 MULTIPOLYGON (((42.74981 -1...
106   Faroe Islands       4562941 MULTIPOLYGON (((-8 63, -6 6...
196           Malta       4810303 MULTIPOLYGON (((14 35.7517,...
411         Vanuatu       5140495 MULTIPOLYGON (((173.526 -18...
201       Mauritius       5546658 MULTIPOLYGON (((64.54151 -1...
202         Mayotte       6190018 MULTIPOLYGON (((44.85604 -1...
306 Solomon Islands       6360777 MULTIPOLYGON (((173.5934 -1...
149   Isle of Wight       7185160 MULTIPOLYGON (((-1.253002 5...
27     Bedfordshire       7289614 MULTIPOLYGON (((-0.295713 5...
```

```
osmextract::geofabrik_zones$name[order(osmextract::geofabrik_zones$pbf_file_size)][1:30]
 [1] "Nauru"                                        
 [2] "Tuvalu"                                       
 [3] "Saint Helena, Ascension, and Tristan da Cunha"
 [4] "Monaco"                                       
 [5] "Niue"                                         
 [6] "Palau"                                        
 [7] "Sao Tome and Principe"                        
 [8] "Cook Islands"                                 
 [9] "Marshall Islands"                             
[10] "Micronesia"                                   
[11] "Rutland"                                      
[12] "Kiribati"                                     
[13] "Andorra"                                      
[14] "Samoa"                                        
[15] "Equatorial Guinea"                            
[16] "Liechtenstein"                                
[17] "Seychelles"                                   
[18] "Enfield"                                      
[19] "Isle of Man"                                  
[20] "Tonga"                                        
[21] "Maldives"                                     
[22] "Comores"                                      
[23] "Faroe Islands"                                
[24] "Malta"                                        
[25] "Vanuatu"                                      
[26] "Mauritius"                                    
[27] "Mayotte"                                      
[28] "Solomon Islands"                              
[29] "Isle of Wight"                                
[30] "Bedfordshire"
```

**Comment by Robinlovelace** [comment]
Based on that another contender could be Enfield: https://www.openstreetmap.org/search?query=enfield#map=12/51.6521/-0.0810

**Comment by natesheehan** [comment]
Apologies for my belated reply here -  I can give either Enfield or IOW a shot, whatever you would think would be more useful?

_On a side,_ I was speaking to Bristol City Council (ex-employer) who loves the project and asked if I'd be able to add some of their data. I'll keep you both posted (and most likely will have a lot of questions).

**Comment by Robinlovelace** [comment]
> Apologies for my belated reply here - I can give either Enfield or IOW a shot, whatever you would think would be more useful?

I vote Enfield as a starter for 10 as it's smaller.

**Comment by Robinlovelace** [comment]
Thanks for looking at this, I expect it will reveal more issues with the `abstr` package so look forward to seeing how it goes. 1st thought: [`ab_scenario()`](https://a-b-street.github.io/abstr/reference/ab_scenario.html) could benefit from being generalised so will be good to see how it gets on with importing desire lines from the PCT. 

Many thanks for chiming in btw @natesheehan !

**Comment by natesheehan** [comment]
@Robinlovelace Using pct to fetch od data for Enfield, few questions:
- is actdev only focusing on cars, walking, cycling od data? e.g. should I exclude bus, train and motorbike?
-Do I also need to add in LSOA regions like you have here https://github.com/ITSLeeds/od/issues/27

`# Aim: prepare data for Enfield (ABSTREET)

library(pct)
library(od)
library(tmap)
library(tidyverse)

# Task 1) Create zones_enfield data
## Get zone data for London using pct
zones = pct::get_pct_zones(region = "london")
## Subset to only include Enfield
enfield_zones = subset(zones,
                       lad_name == c("Enfield"),
                       select = c("geo_code",
                                  "geo_name",
                                  "lad11cd",
                                  "lad_name",
                                  "all",
                                  "bicycle",
                                  "foot",
                                  "car_driver",
                                  "car_passenger",
                                  "motorbike"))

#Task 2) Create desire lines for enfield_desire_lines
## Use pct to fetch OD data
od_london <- pct::get_od(region = "london")
## Subset to only include Enfield
enfield_od = subset(od_london,
                       la_1 == c("Enfield"),
                       select = c("geo_code1",
                                  "geo_code2",
                                  "all",
                                  "train",
                                  "bus",
                                  "bicycle",
                                  "foot",
                                  "car_driver",
                                  "car_passenger",
                                  "motorbike"))

# Fetch centroids for England and wales. Then transform to 4326 spatial projection.
centroids_all <- pct::get_centroids_ew() %>% sf::st_transform(4326)

# Subset to only include Enfield centroids
centroids_enfield <- centroids_all[enfield_zones,]

# Subset and clean Enfield centroid data
centroids_enfield <- centroids_enfield %>%
  select(msoa11cd,geometry) %>%
  rename(
    geo_code = msoa11cd
  )

# Generate desire lines
enfield_desire_lines = od_to_sf(enfield_od, centroids_enfield)

# plot
tmap_mode("plot")
qtm(enfield_zones, borders.lwd = 3) +
  qtm(enfield_desire_lines)
`
![image](https://user-images.githubusercontent.com/22789869/109674106-8b61e600-7b6e-11eb-84b2-85c3decf69ee.png)

**Comment by natesheehan** [comment]
@dabreegster in terms of downloading OSM data, I am using https://extract.bbbike.org/. Is there a specific file format that I should download, or is that handled on the rust side?

**Comment by dabreegster** [comment]
> is actdev only focusing on cars, walking, cycling od data? e.g. should I exclude bus, train and motorbike?

A/B Street does model buses and trains, but currently the route importing from OSM+GTFS is broken, so in practice, all of the trips will degenerate to walking. If they wind up filling out `"mode": "Transit"` in the JSON (I don't recall what `abstr` is doing presently), then it'll eventually work.

> Is there a specific file format that I should download, or is that handled on the rust side?

You need to hand a `.osm` in XML format to the importer. Since this isn't a one-shot (just you) import and we're going to include it permanently, I would prefer geofabrik, since I've had data freshness and network slowness issues with bbike. Once you have the boundary polygon, `cargo run --bin pick_geofabrik your_boundary.poly` will figure out the geofabrik URL for you using their index.

**Comment by Robinlovelace** [comment]
> Do I also need to add in LSOA regions as you have here ITSLeeds/od#27

No I don't think that's necessary. I'm experimenting with the disaggregation function for a separate issue: https://github.com/cyipt/actdev/issues/115

**Comment by Robinlovelace** [comment]
Kind of superseded by #22 so closing for now.
